### PR TITLE
MB-15044 Add a table of contents for the CircleCi Config

### PR DESCRIPTION
**Pull request by YanZ777**
## Summary

The file is kind of giant, so I thought a table of contents would be useful here.
I manually generated the table of contents in the CircleCI config.
The formatter also thought it pertinent to indent a section of comments.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,6 +7,113 @@
 #
 # Please do not use docker_layer_caching! It costs too much money to run. Please set to `false`.
 #
+#
+# Table of Contents
+#
+# refrences:
```
Comment (kctruss) [inline .circleci/config.yml:13]: **typo:** refrences -> references

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,6 +7,113 @@
...
+#  deploy_stg_webhook_client:
+#  deploy_storybook_dp3:
+#  deploy_prd_migrations:
+#  deploy_prd_tasks:
+#  deploy_prd_app:
+#  deploy_prd_app_client_tls:
+#
+# workflows:
+#
+# exeperimental
```
Comment (kctruss) [inline .circleci/config.yml:116]: **typo:** exeperimental -> experimental

**Review by kctruss — COMMENTED** [review]
**polish:** Just spotted a couple of typos

**Review by kctruss — APPROVED** [review]
🌠

**Comment by YanZ777** [comment]
Putting this back in draft since we might not even be using CircleCI in the future (which would eliminate this config.yml).

**Pull request closed without merging**